### PR TITLE
fix resurrect

### PR DESCRIPTION
**Pull request by qndel**
copy/pasted the part from DoHealOther
Resolves https://github.com/diasurgical/devilutionX/issues/2576
int rid would never reach -1 because 
image

### CODE REVIEW

File: Source/spells.cpp [hunk 1]
```diff
@@ -246,14 +246,15 @@ void CastSpell(int id, int spl, int sx, int sy, int dx, int dy, int spllvl)
...
  */
-void DoResurrect(int pnum, int rid)
+void DoResurrect(int pnum, uint16_t rid)
 {
 	if (pnum == MyPlayerId) {
 		NewCursor(CURSOR_HAND);
 	}
 
-	if (rid == -1)
+	if ((DWORD)pnum >= MAX_PLRS || rid >= MAX_PLRS) {
```
Comment (julealgon) [inline Source/spells.cpp:255]: Why would either of these ever be greater than `MAX_PLRS`?

Can't we make `rid` `optional<int>` instead?

Comment (StephenCWills) [reply Source/spells.cpp:255]: Because they're parameters sent via network messages so they need to be validated regardless.

Comment (AJenbo) [reply Source/spells.cpp:255]: We have added a bit of validation for these values, but the more the better before we remove the encryption requirement and implement a game list :hear_no_evil:

**Review by AJenbo — APPROVED** [review]

File: Source/spells.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/spells.cpp:255]: That being said `pnum` should have been validated before this point or we would have OOBs in `ParseCmd()`

Comment (StephenCWills) [reply Source/spells.cpp:255]: FYI, this was previously discussed in #2222.

**Review by StephenCWills — APPROVED** [review]

File: Source/spells.cpp (hunk 1 above)
Comment (qndel) [reply Source/spells.cpp:255]: I just copy/pasted the code from heal other, if you think it's wrong, fix heal other too ;)

**Pull request merged**